### PR TITLE
Ephemeral ES

### PR DESCRIPTION
**Pull request by dzaslavskiy**
ephemeral elasticserach instance for testing

### CODE REVIEW

File: eph-es/audit.sh [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+    hits_count=$(echo "$response" | jq -r '.hits.hits | length')
+
+    while [ "$hits_count" != "0" ]; do
+
+        echo "$response" | jq -c '.hits.hits[]._source'
+
+        response=$(curl -s $es_url/_search/scroll -H "Content-Type: application/json" -d "{ \"scroll\": \"1m\", \"scroll_id\": \"$scroll_id\" }")
+        scroll_id=$(echo "$response" | jq -r ._scroll_id)
+        hits_count=$(echo "$response" | jq -r '.hits.hits | length')
+    done
```
Comment (folksgl) [inline eph-es/audit.sh:81]: If the `echo` line was moved to after the logic here, could we reduce the duplication here a bit? This seems like the same logic:
```suggestion
    hits_count=""

    while [ "$hits_count" != "0" ]; do
        response=$(curl -s $es_url/_search/scroll -H "Content-Type: application/json" -d "{ \"scroll\": \"1m\", \"scroll_id\": \"$scroll_id\" }")
        scroll_id=$(echo "$response" | jq -r ._scroll_id)
        hits_count=$(echo "$response" | jq -r '.hits.hits | length')
        
        echo "$response" | jq -c '.hits.hits[]._source'
    done
```

File: eph-es/deploy.sh [hunk 2]
```diff
@@ -0,0 +1,42 @@
+#!/bin/bash
+
+env=dev
+secrets_file="sk-secrets-dev.json"
+
+echo "press key to continue to next step:"
+read -pr "1. Deploy elasticsearch app"
```
Comment (folksgl) [inline eph-es/deploy.sh:7]:
```suggestion
# The 'read' commands here serve as a way to pause execution of the
# script until the user is ready to continue.
echo "press key to continue to next step:"
read -pr "1. Deploy elasticsearch app"
```

File: eph-es/deploy.sh [hunk 3]
```diff
@@ -0,0 +1,42 @@
+#!/bin/bash
+
+env=dev
+secrets_file="sk-secrets-dev.json"
```
Comment (folksgl) [inline eph-es/deploy.sh:4]: Is hardcoding `dev` intentional?

File: eph-es/deploy.sh [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+read -pr "2. Network policies"
+
+cf add-network-policy sk-esconfigs eph-es --protocol tcp --port 61443
+cf add-network-policy sk-events eph-es --protocol tcp --port 61443
+cf add-network-policy sk-events-read eph-es --protocol tcp --port 61443
+cf add-network-policy sk-analytics eph-es --protocol tcp --port 61443
+
+read -pr "3. configure secrets"
+
+jq --arg env "$env" '.es.usePath = false | .es.url = "identity-idva-es-$env.apps.internal:61443" | .es.accessKeyId = "id" | .es.secretAccessKey = "sec"' "$secrets_file" > sk-secrets-es.json
```
Comment (folksgl) [inline eph-es/deploy.sh:20]: This JQ command could use a comment describing its effect.

File: eph-es/manifest.yaml [hunk 5]
```diff
@@ -0,0 +1,25 @@
+---
+applications:
+  - name: eph-es
+    routes:
+      - route: identity-idva-es-((ENVIRONMENT)).apps.internal
+    health-check-type: process
```
Comment (folksgl) [inline eph-es/manifest.yaml:6]: Is this applicable to our cloud.gov ES app? https://www.elastic.co/guide/en/elasticsearch/reference/current/cluster-health.html

**Review by folksgl — CHANGES_REQUESTED** [review]
The usage of `eph` here unfortunately only has meaning to those of us who have been hearing the term "ephemeral elasticsearch" over the last few weeks. We should likely use the term `temp-es`, `debug-es`, or the longer `ephemeral-es`.

File: eph-es/restore.sh [hunk 6]
```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+
+secrets_file="sk-secrets-dev.json"
```
[the remark this answers is not in the document]
Comment (dzaslavskiy) [reply eph-es/restore.sh:3]: it just an example

File: eph-es/deploy.sh (hunk 3 above)
Comment (dzaslavskiy) [reply eph-es/deploy.sh:4]: as above

**Comment by dzaslavskiy** [comment]
I quite attached to the name eph-es (pronounced as 'f s'). In which context would you say additional info is needed?

File: eph-es/audit.sh [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+                    {
+                        "match_all": {}
+                    }
+                ],
+                "should": [],
+                "must_not": []
+            }
+        },
+        "search_after": [0],
+        "sort" : [ "tsEms"]
```
[the remark this answers is not in the document]
Comment (dzaslavskiy) [reply eph-es/audit.sh:124]: It probably possible to do something clever here and not have two similar queries but I am not sure it is worth it. They ultimately work differently.

File: eph-es/manifest.yaml (hunk 5 above)
Comment (dzaslavskiy) [reply eph-es/manifest.yaml:6]: sure. es does listen on http so we could use http health check on any path. i don't see a big differance.

**Comment by folksgl** [comment]
> I am quite attached to the name 'eph-es' (pronounced as 'f s'). In which context would you say additional info is needed?

I'm trying to think ahead to someone who isn't us trying to read/understand the code. The readme uses the term "Ephemeral Elasticsearch" in multiple places so I guess it's not too much of a puzzle for a new dev should we get a new contributor.

File: eph-es/manifest.yaml (hunk 5 above)
Comment (dzaslavskiy) [reply eph-es/manifest.yaml]: i moved it to a port check.

**Review by folksgl — APPROVED** [review]

**Pull request merged**